app/sweepers.py

```diff
@@ -0,0 +1,76 @@
+
+"""
+This module contains the definitions of Sagittariidae's sweeper processes.
+These are asynchronous processes that designed to drive the current state of
+the world to a desired final state.
+
+The sweepers are run as processes separate from the webservice.  They may be
+invoked by name by executing this module, typically from a scheduler such as
+`cron`.
+"""
+
+import os
+import shutil
+import sys
+
+import app as sagittariidae
+import models
+
+
+logger = sagittariidae.app.logger
+
+class Sweeper(object):
+
+    def run(self):
+        raise NotImplementedError()
+
+    def sweep(self):
+        try:
+            self.run()
+        except Exception, e:
+            logger.error('Unhandled exception in sweeper %s', self, exc_info=e)
```

Comment:
It would violate the unique constraint, so a re-upload isn't possible.  At this point I don't think it's worth opening the door to re-uploads and putting the fileset immutability at risk.  This should be something of an exceptional case, because to have gotten this far we will have validated the upload and the reconstruction of the file.  The most likely problem that I can think of is something permissions or file-system-related, in which case a re-upload wouldn't help anyway.  I expect it's going to be the kind of thing that we debug by hand a couple of times, and add code for as we identify resolution.